public/javascripts/hello.js

```diff
@@ -1,3 +1,97 @@
 const es6Hello = "Hello OpenSD";
 
 console.log(es6Hello);
+
+var repoElements = Array.from(document.querySelectorAll("div.repository-selection button"));
+repoElements.forEach(function(el, i) {
+  el.addEventListener("click", function() {
+    el.classList.toggle("active");
+  });
+});
+
+var searchListeners = Array.from(document.querySelectorAll("div.main-search"));
+searchListeners.forEach(function(el, i){
+  el.addEventListener("click", function(){
+  //add button
+    if(event.target.classList.contains('add-search-button')){
+      var searchBarNodeClone = document.querySelectorAll('div.reusable-search')[0].cloneNode(true);
+      searchBarNodeClone.childNodes[0].childNodes[0].innerHTML = "Select " + "<span class=\"caret\"></span>";
```

Comment:
If searchBarNodeClone does not have any children this code will crash which could be catastrophic.